calico_node/allocate_ipip_addr/allocate_ipip_addr.go

```diff
@@ -0,0 +1,112 @@
+package allocateIPIPAddr
+
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"debug/pe"
+)
+
+func allocateIPIPAddr(client *client.Client, string nodename) {
+	meta := api.IPPoolMetadata{}
+	ipPoolList, err := client.IPPools().List(meta)
+	if err != nil {
+		log.Warnf("error retrieving pools: %s", err)
+	}
+    ipv4Pools := getIPv4Pools(ipPoolList.Items)
```

Comment:
Mixing tabs and spaces here.   Make sure you're using an IDE that supports go.

I'm using PyCharm with a golang plugin.